Adds the Vox Slugsling

Kearel opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Adds the slugsling, a vox specific weapon available in Heist that fires an egg. Depending on the mode chosen by the weapon, the egg may hatch on impact or when somebody passes by.

Inside the egg is a slug, a viscous, many teethed creature with a penance for attaching itself to people. When attached, the slug injects a steady amount of synapsis into the player, causing a slow but steady buildup of chems.

Slugs have 15 hp and are very small. They inject 0.5 units of synaps per tick and attach themselves depending on the brute damage of the person in question.

Also fixes a miscelanoius issue where launchers do not play their fire sound, as well as add a base vox weapon type for future weapons and such and such.

Kearel committed:
Adds Slugsling

Kearel committed:
Debug code woops

ThatOneGuy42 commented:
I absolutely love this idea

On Feb 1, 2017 2:08 PM, "Kearel" <notifications@github.com> wrote:

> Adds the slugsling, a vox specific weapon available in Heist that fires an
> egg. Depending on the mode chosen by the weapon, the egg may hatch on
> impact or when somebody passes by.
>
> Inside the egg is a slug, a viscous, many teethed creature with a penance
> for attaching itself to people. When attached, the slug injects a steady
> amount of synapsis into the player, causing a slow but steady buildup of
> chems.
>
> Slugs have 15 hp and are very small. They inject 0.5 units of synaps per
> tick and attach themselves depending on the brute damage of the person in
> question.
>
> Also fixes a miscelanoius issue where launchers do not play their fire
> sound, as well as add a base vox weapon type for future weapons and such
> and such.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Baystation12/Baystation12/pull/15945
> Commit Summary
>
>    - Initial commit
>    - Second commit
>    - Adds changelog
>
> File Changes
>
>    - *M* baystation12.dme
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-0>
>    (4)
>    - *M* code/modules/item_worth/worths_list.dm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-1>
>    (2)
>    - *A* code/modules/mob/living/simple_animal/hostile/voxslug.dm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-2>
>    (94)
>    - *M* code/modules/projectiles/gun.dm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-3>
>    (13)
>    - *D* code/modules/projectiles/guns/alien.dm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-4>
>    (54)
>    - *M* code/modules/projectiles/guns/launcher.dm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-5>
>    (1)
>    - *A* code/modules/projectiles/guns/launcher/alien.dm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-6>
>    (59)
>    - *A* code/modules/projectiles/guns/launcher/slugsling.dm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-7>
>    (67)
>    - *A* html/changelogs/TheWelp - SlugSling.yml
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-8>
>    (37)
>    - *M* icons/mob/animal.dmi
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-9>
>    (0)
>    - *M* icons/mob/items/lefthand_holder.dmi
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-10>
>    (0)
>    - *M* icons/mob/items/righthand_holder.dmi
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-11>
>    (0)
>    - *M* icons/obj/gun.dmi
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-12>
>    (0)
>    - *M* icons/obj/weapons.dmi
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-13>
>    (0)
>    - *M* maps/torch/torch-10.dmm
>    <https://github.com/Baystation12/Baystation12/pull/15945/files#diff-14>
>    (4)
>
> Patch Links:
>
>    - https://github.com/Baystation12/Baystation12/pull/15945.patch
>    - https://github.com/Baystation12/Baystation12/pull/15945.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Baystation12/Baystation12/pull/15945>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/ABYOUUPSf-i-mk-cO-Y5mxSRvpbwtJpuks5rYOYdgaJpZM4L0U2q>
> .
>

Kearel commented:
Debug code caused problem.

Kearel commented:
All credit goes to @masterrbc. He helped me make this.

Techhead0 commented on code/modules/mob/living/simple_animal/hostile/voxslug.dm:
```diff
@@ -0,0 +1,94 @@
+			to_chat(H, "<span class='warning'>You feel strange as \the [src] pulses...</span>")
+		var/datum/reagents/R = H.reagents
+		R.add_reagent("synaptizine", 0.5)
+		return 1
```
I have a few concerns with this codeblock:
- The codeblock will fire if a friendly Vox scoops them up.
- No guarantee `src.loc.loc` will be a human.
- Seemingly no need for `H` to be cast as a human.
- Avoiding [this bit](https://github.com/Baystation12/Baystation12/blob/74b6687fe7d92f35c2b95c143e16370b33298c29/code/modules/mob/living/simple_animal/simple_animal.dm#L73-L92) just because you're inside someone is pretty sketch.

Kearel replied:
I'm not avoiding that bit, I'm avoiding [this](https://github.com/Baystation12/Baystation12/blob/dev/code/modules/mob/living/simple_animal/hostile/hostile.dm#L134) bit.

Kearel replied:
You know what, I think you're right, the more I think about it the better it'd be if I make sure it DOES check that stuff, just for some future proofing.

I'll make a new stance for hostile animals that basically says 'you are inside an object/mob so dont do anything' and when they aren't inside something they'll jsut switch stance

Techhead0 replied:
The problem is that it avoids all the bits. The bit I linked is one that should not be avoided.

Kearel replied:
Alright, I fixed that, now I have to figure out a simple way to check if something is embeded...

Kearel replied:
I guess I'll just check implants. Since it should realistically be the fastest way of doing it.

Kearel commented on code/modules/mob/living/simple_animal/hostile/voxslug.dm:
```diff
@@ -0,0 +1,94 @@
+			to_chat(H, "<span class='warning'>You feel strange as \the [src] pulses...</span>")
+		var/datum/reagents/R = H.reagents
+		R.add_reagent("synaptizine", 0.5)
+		return 1
```
Another thing that is kinda relavent, if inside a holder, the mob shouldn't be able to have their health/state changed. Meaning that there really isn't a reason to change stat since it should theoretically always be the same (living/dead).

The scoop thing is weird though, yeah, I'll get on that.

Kearel committed:
Fixes bugs. Adds logic to hostiles being in things as well.

Kearel commented:
Alright I think I did what you asked, Techhead. I made it so that it always does ..() before doing anything. I also added a stance for hostiles that just makes them not do anything (scan for enemies, walk, etc) if they are inside an object/mob.

Also fixed Exodus's paths.

Kelenius commented:
I'm not really sure what is the point of shooting a slug that gives your enemy a painkiller and also makes them recover from stuns quickly. Synaptizine's toxicity is VERY mild, and the overdose would take a while to reach, and still isn't that dangerous.

Kearel commented:
You can also attach them to people, so you could use them as well. Maybe instead of just shooting in synap, convert blood to synap? that way it could be used offensively (if there are multiple on a person) and defensively.

Raptor1628 commented on code/modules/mob/living/simple_animal/hostile/voxslug.dm:
```diff
@@ -0,0 +1,93 @@
+
+/mob/living/simple_animal/hostile/voxslug
+	name = "slug"
+	desc = "A viscious little creature, it has a mouth of too many teeth and a penance for blood."
```
*penchant for blood

Kearel replied:
Nice catch.

mwerezak commented:
@Kearel I think you should decide if they are meant to be offensive or defensive and stick to it because the item does seem rather unfocused at the moment. Given how the launcher works, I'd suggest focusing on being offensive and make the slugs do nasty things to people, like drain their blood and inject bad chems, not medicines like synaptizine.

Kearel commented:
Probably a good idea.

Kearel committed:
Slug now injects woozifying drugs. Good luck.

Kearel commented:
Anything else I have to change?

ghost commented:
I just saw this and I gotta say that's an interesting edit of my gun sprite. I like how making some randomly shaped gun is able to be used in such different ways across code-bases.

Kearel commented:
It was bulbous af so it was PERFECT for what I had in mind.
